### PR TITLE
LeaveOpen ctor for CryptoStream

### PR DESCRIPTION
**Pull request by bbowyersmyth**
Resolves https://github.com/dotnet/corefx/issues/1079

cc @bartonjs

### CODE REVIEW

File: src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/CryptoStream.cs [hunk 1]
```diff
@@ -25,15 +25,22 @@ public class CryptoStream : Stream, IDisposable
         private bool _canWrite;
         private bool _finalBlockTransformed;
         private SemaphoreSlim _lazyAsyncActiveSemaphore;
-        
+        private bool _leaveOpen;
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/CryptoStream.cs]: Nit: readonly

Comment (bbowyersmyth) [inline src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/CryptoStream.cs]: Done

**Comment by bartonjs** [comment]
@bbowyersmyth this will require revving the contract version, and changing it to target netcoreapp1.1 (instead of netstandard1.x).  There's an in-progress of that kind of change at #11272; but we don't have a good example where we've actually committed one yet.  So we might need to hold off a couple more days while we get a reference example committed :smile:

**Comment by stephentoub** [comment]
@bartonjs, can we move this one forward now?

**Comment by bartonjs** [comment]
We haven't merged a netstandard20 change into Cryptography.Primitives yet, so this either needs to wait a little longer to piggy back on #11959 or do all the netstandard1.7 related bumps in this change.  (Or, a third option, just bump it in change by itself)

**Comment by bartonjs** [comment]
Okay, the change to restructure the ref assembly for S.S.C.Algorithms has gone in now.  I don't know if this new member should go in netstandard1.7 or needs to be forked off as netcoreapp11 (or netcoreapp12?).  Do you happen to know, @stephentoub? :smile:

**Comment by Petermarcu** [comment]
@ericstj , can you confirm? In general, things going into master right now go into netcoreapp1.2.

**Comment by ericstj** [comment]
> @ericstj , can you confirm? In general, things going into master right now go into netcoreapp1.2.

If the API already exists in desktop it's goes into netstandard1.7 (until we get a netstandard2.0 mapped by NuGet).

If the API is brand-new it should go into netcoreapp*.

**Comment by ericstj** [comment]
To be clear, things in master right now are targeting netstandard1.7 / netcoreapp1.1 as a stopgap until we move them to netstandard2.0 / netcoreapp1.2.

**Comment by karelz** [comment]
#11959 is now merged. What should happen with this PR?
cc: @steveharter

**Comment by bartonjs** [comment]
I'm cherry-picking the current commits (though squashed into one) and adding the contract change, and will likely replace this PR shortly.

**Comment by bartonjs** [comment]
Solved merge conflict and added contract work in #12718, closing this PR as it has been replaced.

**Pull request closed without merging**